src/app-logic/url-handling.js

```diff
@@ -1247,3 +1266,44 @@ function validateTimelineType(type: ?string): TimelineType {
       return 'cpu-category';
   }
 }
+
+/**
+ * Parses the value of the `assemblyView` parameter in the URL.
+ * This parameter has the following form:
+ *   <libIndex> '~' <hexAddress> '~' <hexSize> '_'? '~' <symbolName>
+ */
+function parseAssemblyViewSymbol(value: string): NativeSymbolInfo | null {
+  const [libIndexStr, hexAddress, sizeStr, name] = value.split('~');
+  const libIndex = parseInt(libIndexStr, 10);
+  const address = parseInt(hexAddress, 16);
+  // sizeStr is `b7c` or `b7c_`, the trailing underscore means "or more".
+  const [functionSizeStr, functionSizeIsKnown] = sizeStr.endsWith('_')
+    ? [sizeStr.substr(0, sizeStr.length - 1), false]
```

Comment:
nit: `substr` is not recommended to use nowadays. Better use `slice` that also accepts negative indices to count from the end of the string which makes this line simpler:
```
sizeStr.slice(0, -1)
```

Reply:
Oh I keep for getting about slice!